crates/index-scheduler/src/scheduler/mod.rs

```diff
@@ -339,6 +341,28 @@ impl IndexScheduler {
 
         // We must re-add the canceled task so they're part of the same batch.
         ids |= canceled;
+
+        processing_batch.stats.call_trace =
+            progress.accumulated_durations().into_iter().map(|(k, v)| (k, v.into())).collect();
+        processing_batch.stats.write_channel_congestion = congestion.map(|congestion| {
+            let mut congestion_info = serde_json::Map::new();
+            congestion_info.insert("attempts".into(), congestion.attempts.into());
+            congestion_info.insert("blocking_attempts".into(), congestion.blocking_attempts.into());
+            congestion_info.insert("blocking_ratio".into(), congestion.congestion_ratio().into());
+            congestion_info
+        });
+
```

Comment:
Why don’t we define a type if it’s always the same entries?

Reply:
We don't want to do that as these structures could change in the future, and using a hardly typed structure will break a dump-less upgrade for no real added value.